xbmc/guilib/TextureGL.cpp

```diff
@@ -65,31 +72,38 @@ void CGLTexture::LoadToGPU()
   }
 
   // Bind the texture object
-  glBindTexture(GL_TEXTURE_2D, m_texture);
+  gl::BindTexture(GL_TEXTURE_2D, m_texture);
 
   GLenum filter = (m_scalingMethod == TEXTURE_SCALING::NEAREST ? GL_NEAREST : GL_LINEAR);
 
+#ifdef HAS_GL
+  auto renderSystemGL = dynamic_cast<CRenderSystemGL*>(CServiceBroker::GetRenderSystem());
```

Comment:
About here looks a bit a strange, as `CServiceBroker::GetRenderSystem()` sound global and relates to both (GLES / GL).
Maybe a function e.g. `CServiceBroker::SupportsRenderSystem(GL_RENDERING)` or `CServiceBroker::SupportsGLRenderSystem()`?

The related added function can maybe still use `dynamic_cast<CRenderSystemGL*>` if it works (is new for me that it can give nullptr if class not match, need still to learn :sweat_smile:). Then outside places are more clean if by a function.

Reply:
dynamic_cast can return `nullptr` yes, that's why it's used this way. I would like in the future to add abstraction by splitting the GL and GLES code paths into separate classes. This is similar to what is done in #15549